bowtie/_cli.py

```diff
@@ -714,19 +716,69 @@ def _check_fail_fast_provided(
     metavar="IMPLEMENTATION",
     help="A docker image which implements the bowtie IO protocol.",
 )
-DIALECT = click.option(
-    "--dialect",
-    "-D",
-    "dialect",
-    type=_Dialect(),
-    default=max(Dialect.known()),
-    show_default=True,
-    metavar="URI_OR_NAME",
-    help=(
-        "A URI or shortname identifying the dialect of each test. Possible "
-        f"shortnames include: {', '.join(sorted(Dialect.by_alias()))}."
+
+
+def create_dialect_option():
+    def wrapper(default: Any | None = max(Dialect.known())):
+        return click.option(
+            "--dialect",
+            "-D",
+            "dialect",
+            type=_Dialect(),
+            default=default,
+            show_default=True,
+            metavar="URI_OR_NAME",
+            help=(
+                "A URI or shortname identifying the dialect of each test. "
+                "Possible shortnames include: "
+                f"{', '.join(sorted(Dialect.by_alias()))}."
+            ),
+        )
+
+    return wrapper
+
+
+DIALECT = create_dialect_option()
+LANGUAGE = click.option(
```

Comment:
Please never hardcode things, that should set off red alarm bells.

Here a simple way to get what you need here presumably is to use `Implementation.known` and then grab the stuff before the hyphen, unless we think of an even better idea, but that seems pretty ok to me for now. You could also do some sort of aliasing similar to what we do in the frontend, so that someone can type `typescript` instead of `ts`, but yeah, no need to add something which will so easily go stale.

Reply:
Will do these changes. Thanks!

Reply:
No need to be sorry! You definitely don't *offend* or anything :D, I'm mentioning these things in the hope they're useful to learn! But apologies are never needed for not knowing something!